include/cetl/any.hpp

```diff
@@ -294,10 +316,84 @@ class any : private detail::base_move<Footprint, Copyable, Movable>
 template <typename ValueType, typename Any, typename... Args>
 CETL_NODISCARD Any make_any(Args&&... args)
 {
-    return Any(cetl::in_place_type<ValueType>, std::forward<Args>(args)...);
+    return Any(in_place_type<ValueType>, std::forward<Args>(args)...);
+}
+
+/// \brief Constructs an any object containing an object of type T, passing the provided arguments to T's constructor.
+///
+/// Equivalent to `cetl::any(cetl::in_place_type<ValueType>, list, std::forward<Args>(args)...)`.
+///
+template <typename ValueType, typename Any, typename Up, typename... Args>
```

Comment:
I wonder if we could default Any to something like this:

```suggestion
template <T>
using any_like = any<sizeof(T),
                    std::is_copy_constructible<T>::value,
                    std::is_move_constructible<T>::value,
                    std::max(alignof(T), alignof(std::max_align_t))>;

template <typename ValueType, typename Any = any_like<ValueType>, typename Up, typename... Args>
```

I only thought about this now but it should have been during the previous review.

Do you think it makes sense? Otherwise we deviate from the semantics of std::make_any quite a bit as it doesn't require one to specify the type of any.